core/include/scipp/core/element/arithmetic.h

```diff
@@ -22,8 +22,33 @@ constexpr auto add_inplace_types =
              std::tuple<float, int64_t>, std::tuple<float, int32_t>,
              std::tuple<int64_t, bool>>;
 
+constexpr auto add_inplace_nan_types =
+    arg_list<double, float, int64_t, int32_t, std::tuple<double, float>,
+             std::tuple<float, double>, std::tuple<int64_t, int32_t>,
+             std::tuple<int32_t, int64_t>, std::tuple<double, int64_t>,
+             std::tuple<double, int32_t>, std::tuple<float, int64_t>,
+             std::tuple<float, int32_t>, std::tuple<int64_t, bool>>;
+
 constexpr auto plus_equals =
     overloaded{add_inplace_types, [](auto &&a, const auto &b) { a += b; }};
+constexpr auto nan_plus_equals =
+    overloaded{add_inplace_nan_types, [](auto &&a, const auto &b) {
+                 using ElementType = std::decay_t<decltype(a)>;
+                 auto constexpr zero = []() {
+                   if constexpr (is_ValueAndVariance_v<ElementType>)
+                     return ElementType{0, 0};
+                   else
+                     return ElementType{0};
+                 }();
+                 using std::isnan;
+                 if (isnan(a))
+                   a = zero;
```

Comment:
Does `a = std::decay_t<decltype(a)>{};` work?

Reply:
Unfortunately not, because such constructor for ValueAndVariance, but I think it would be reasonable to add one.

Reply:
I would have thought the default constructor should exist (compiler generated?)?

Reply:
No because non-default constructors are defined. However I note that there is a constructor that should allow us to do the above. https://github.com/scipp/scipp/blob/master/core/include/scipp/core/value_and_variance.h#L28. i.e. 
`a = std::decay_t<decltype(a)>{0}` should work.